core/src/avm2/globals/flash/display/shader_job.rs

```diff
@@ -178,6 +172,52 @@ pub fn get_shader_args<'gc>(
     Ok((shader_handle.clone(), args))
 }
 
+trait PixelSource {
+    fn collect<const N: usize>(&self, num_pixels: usize) -> Option<Vec<[f32; N]>>;
+}
+
+impl PixelSource for &[f32] {
+    fn collect<const N: usize>(&self, num_pixels: usize) -> Option<Vec<[f32; N]>> {
+        let (floats, _) = self.as_chunks::<N>();
+        Some(floats.get(..num_pixels)?.to_vec())
+    }
+}
+
+impl<'gc> PixelSource for &[Value<'gc>] {
+    fn collect<const N: usize>(&self, num_pixels: usize) -> Option<Vec<[f32; N]>> {
+        let (chunks, _) = self.as_chunks::<N>();
+        Some(
+            chunks
+                .get(..num_pixels)?
+                .iter()
+                .map(|vals| vals.map(|val| val.as_f64() as f32))
+                .collect(),
+        )
+    }
+}
+
+fn make_float_texture<'gc, S: PixelSource>(
+    source: S,
+    width: u32,
+    height: u32,
+    input_channels: u32,
+    mut err: impl FnMut() -> Error<'gc>,
```

Comment:
It's better to pass required arguments (activation, name) instead of a lambda, because (1) the error is the same in both cases, so no duplication, (2) if someone needs activation in the future for something here, it wouldn't be possible to borrow it twice.

It's fine to define `err` in this function.